exporters/prometheus/exporter.go

```diff
@@ -233,3 +233,20 @@ func sanitizeRune(r rune) rune {
 	}
 	return '_'
 }
+
+// sanitizeName replaces characters which don't match the regex `^[a-zA-Z_:][a-zA-Z0-9_:]*$` with `_`.
+func sanitizeName(n string) string {
+	if len(n) == 0 {
+		return n
+	}
+	var sn = make([]string, len(n))
+	for i, b := range n {
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || b == ':' || (b >= '0' && b <= '9' && i > 0)) {
```

Comment:
If the first rune is [0-9] it needs to be prefixed instead of replaced: https://github.com/open-telemetry/opentelemetry-specification/blob/ca593258953811eb33197976b8e565299b874853/specification/metrics/semantic_conventions/openmetrics-guidelines.md#name-and-attribute-keys

Reply:
That is already the case. The last check in the condition only returns digits to be valid if they aren't the first character.

Reply:
Sure, but it would replace that digit not prefix it.